### PR TITLE
Npm

### PR DESCRIPTION
**Pull request by brettz9**
Builds on PR #645

- npm: Update deps (jsdom, xmldom) and devDeps
- npm: Add eslint to build script
- Refactoring: Consistent quotes (single); unused vars.
- Refactoring: Use new jsdom APIs (and further prepare for use of canvas API)
- Docs: Succincter heading style, avoid script type

### CODE REVIEW

File: test/_generate_expected_node.js [hunk 1]
```diff
@@ -1,30 +1,85 @@
-const canvg = require("../src/canvg.js"),
```
Comment (gabelerner) [inline test/_generate_expected_node.js]: can we remove this file from this PR? it's in another one

File: src/canvg.js [hunk 2]
```diff
@@ -3,32 +3,33 @@
 /* eslint-disable no-redeclare */
 var RGBColor = require('rgbcolor'),
   StackBlur = require('stackblur-canvas'),
-  Canvas = require("canvas");
```
Comment (gabelerner) [inline src/canvg.js]: changes in this file look good

File: package.json [hunk 3]
```diff
@@ -50,10 +50,10 @@
   ],
   "readmeFilename": "README.md",
   "dependencies": {
-    "jsdom": "^8.1.0",
+    "jsdom": "^13.0.0",
```
[the remark this answers is not in the document]
Comment (brettz9) [reply package.json]: Hmm... Interesting I didn't come across issues in testing, though maybe in switching around, I still was using a copy of canvas 2.0....

I reverted all package.json changes related to dependencies, but kept two which I presume should be fine by you to merge now?

This PR also includes changes to the jsdom API, however, so I assume I should revert those portions too? And do you also want me to revert the build file updates? Those updates surfaced your changes removing the bitwise operators, so I'd assume they'd be safe...

And by "that PR", are you referring to one you are working on to get things updated to canvas 2.0? If so, let me know when done, so I can reapply the other dep updates, unless you are planning to handle that as well..

[the remark this answers is not in the document]
Comment (gabelerner) [reply package.json]: had some time so i started the PR @ https://github.com/canvg/canvg/pull/652 which has some of your changes ... once that's merged we'll see what's left in this one

**Comment by gabelerner** [comment]
@brettz9 i backmerged master in and kept the rest of your changes. we're now targeting a 2.0.0 release of canvg against canvas 2.x ... ready to merge this PR?

**Comment by gabelerner** [comment]
also tests should now be golden. travis only runs node and it'll fail if there's a regression

**Comment by brettz9** [comment]
The merging process seems good, and if you're evading browser tests for good reason, that's fine, but:

1. Running `npm test` (or `npm run test-node`) on my Mac with Node 9.2.1 (via nvm) gives me a quick exit now with a `SIGSEGV`, and on Node 10.11.0 (nvm) on the same machine, the tests execute but 4 tests fail (`issue88.svg`, `issue593.svg`, `36.svg`, `issue342.svg`).
2. I don't remember the browser tests failing before (and now 22/148 are failing (regardless of Node version)).

**Comment by brettz9** [comment]
On the plus side, with Node 10.11.0, `generate-expected` got through all files without problem, and when re-running the tests against these generated files, I get (with v10.11.0) 0 Node tests failing (though still 22 browser tests failing).

But I have different generated PNGs which I haven't added to this PR.

**Comment by gabelerner** [comment]
Unless we dockerize the test suite, every computer will be something different which is why there's a threshold of `0.03` in `blink-diff`. This gets most computers close to passing (my Mac fully passes node on master) and we should aim for Travis passing with correct expected images (should do a visual audit at some point) vs. getting local to pass.

Since Travis runs only node tests now, I reneabled browser to fail hard.  That also will be different results based on local settings.  I'll go through at some point and try to clean up to see if there are real issues, but my initial thoughts are it's just fonts and how the browser deals with external images.

**Comment by brettz9** [comment]
Ok, sure... SGTM... So if you've already run and committed the results of `generate-expected`, I'd say it's ready for merging...

**Pull request merged**